certora/specs/ConsistentState.spec

```
@@ -41,3 +45,112 @@ invariant supplyQueueLengthInRange()
 
 invariant withdrawQueueLengthInRange()
     withdrawQueueLength() <= maxQueueLength();
+
+function hasNoBadPendingTimelock() returns bool {
+    uint192 pendingValue;
+    uint64 validAt;
+    pendingValue, validAt = pendingTimelock();
+
+    return validAt == 0 <=> pendingValue == 0;
+}
+
+invariant noBadPendingTimelock()
+    hasNoBadPendingTimelock()
+{
+    preserved with (env e) {
+        requireInvariant timelockInRange();
+        // Safe require as it corresponds to year 2262.
+        require e.block.timestamp < 2^63;
+    }
+}
+
+function isDifferentPendingTimelock() returns bool {
+    uint192 pendingValue;
+    pendingValue, _ = pendingTimelock();
+
+    return assert_uint256(pendingValue) != timelock();
+}
+
+invariant differentPendingTimelock()
+    isDifferentPendingTimelock()
+{
+    preserved {
+        requireInvariant pendingTimelockInRange();
+        requireInvariant timelockInRange();
+    }
+}
+
+function isGreaterPendingTimelock() returns bool {
+    uint192 pendingValue;
+    pendingValue, _ = pendingTimelock();
+
+    return assert_uint256(pendingValue) != 0 => assert_uint256(pendingValue) != timelock();
+}
```

- a timelock is pending iff it is set to a lower value right? So it can only be pending it is smaller
- here is is returned pendingValue!=timelock and not pendingValue<timelock
```suggestion
function isGreaterPendingTimelock() returns bool {
    uint192 pendingValue;
    pendingValue, _ = pendingTimelock();

    return assert_uint256(pendingValue) < timelock();
}
```